momepy/preprocessing.py

```diff
@@ -1101,3 +1105,453 @@ def roundabout_simplification(
+    )
+    cluster_geometries = gpd.points_from_xy(
+        cluster_centroids_df[x_col], cluster_centroids_df[y_col]
+    )
+    cluster_gdf = gpd.GeoDataFrame(
+        cluster_centroids_df, crs=graph_crs, geometry=cluster_geometries
+    )
+    cluster_nodes_list = list(cluster_gdf.to_dict("index").items())
+
+    # Rebuild edges if necessary:
+    if rebuild_graph:
+        graph.graph["approach"] = "primal"
+        edges_gdf = nx_to_gdf(graph, points=False, lines=True)
+        simplified_edges = _get_rebuilt_edges(
+            edges_gdf,
+            nodes_to_merge_dict,
+            cluster_gdf,
+            method=rebuild_edges_method,
+            buffer=1.5 * tolerance,
+            edge_from_col=edge_from_col,
+            edge_to_col=edge_to_col,
+        )
+
+    # Replacing the collapsed nodes with centroids and adding edges:
+    simplified_graph = graph.copy()
+    if not directed:
+        simplified_graph = nx.MultiGraph(simplified_graph)
+    simplified_graph.remove_nodes_from(nodes_to_merge_df.index)
+    simplified_graph.add_nodes_from(cluster_nodes_list)
+
+    if rebuild_graph:
+        simplified_graph.add_edges_from(simplified_edges)
+
+    return simplified_graph
+
+
+def _get_rebuilt_edges(
+    edges_gdf,
+    nodes_dict,
+    cluster_gdf,
+    method="spider",
+    buffer=45,
+    edge_from_col="from",
+    edge_to_col="to",
+):
+    """
+    Update origin and destination on network edges when original endpoints were replaced by a
+      consolidated node cluster. New edges are drawn according to method which is one of:
+
+    1. Extension reconstruction:
+        Edges are linearly extended from original endpoints until the new nodes. This method preserves
+        most faithfully the network geometry.
+    2. Spider-web reconstruction:
+        Edges are cropped within a buffer of the new endpoints and linearly extended from there. This
+        method improves upon linear reconstruction by mantaining, when possible, network planarity.
+    3. Euclidean reconstruction:
+        Edges are ignored and new edges are built as straightlines between new origin and new
+        destination. This method ignores geometry, but efficiently preserves adjacency.
+
+    Parameters
+    ----------
+    edges_gdf : GeoDataFrame
+        GeoDataFrame containing LineString geometry and columns determining origin
+        and destination node ids
+    nodes_dict: dict
+        Dictionary whose keys are node ids and values are the corresponding consolidated
+        node cluster ids. Only consolidated nodes are in the dictionary.
+    cluster_gdf : GeoDataFrame
+        GeoDataFrame containing consolidated node ids.
+    method: string
+        'extension' or 'spider' or 'euclidean'
+    buffer : float
+        distance to buffer consolidated nodes in the Spider-web reconstruction
+    edge_from_col, edge_to_col: string
+        edge attribute with the valid origin/destination node id
+
+    Returns
+    ----------
+    List
```

Comment:
What's the name of this variable?

Reply:
I think I did not properly name output variables. I usually include in the function description a brief explanation of the output and then just give the type below the Returns block. Should I arbitrarily name it based on what it represents?

Reply:
I don't think so, it is fine as it is imho. That applies to all cases of this.

Reply:
My personal preference is explicit variable naming for descriptive docstrings, but I will defer the @martinfleis's decision on this.